src/hotspot/share/opto/graphKit.cpp

```diff
@@ -1970,13 +1990,18 @@ Node* GraphKit::set_results_for_java_call(CallJavaNode* call, bool separate_io_p
   }
 
   // We just called the constructor on a value type receiver. Reload it from the buffer
-  if (call->method()->is_object_constructor() && call->method()->holder()->is_inlinetype()) {
```

Comment:
`call->method()->holder()->is_inlinetype()` false for abstract value classes -> use bottom type of receiver instead.